classes/client/s3_client.php

```diff
@@ -80,10 +80,34 @@ public function __wakeup() {
     }
 
     public function set_client($config) {
+        $proxy = '';
+        if (!empty($config->proxyserver)) {
+            if (!empty($config->proxyusername)) {
+                $arrayproxy = explode("//", $config->proxyserver);
+                if (count($arrayproxy) > 2) {
+                    $proxy = $arrayproxy[0]
+                        . "//" . $config->proxyusername
+                        . ":" . $config->proxypassword
+                        . "@" . $arrayproxy[1]
+                        . ":" . $config->proxyport;
+                } else {
+                    $proxy = $config->proxyusername
+                        . ":" . $config->proxypassword
+                        . "@" . $arrayproxy[0]
+                        . ":" . $config->proxyport;
+                }
+            } else {
+                $proxy = $config->proxyserver.':'.$config->proxyport;
+            }
+        }
+
         $this->client = S3Client::factory(array(
         'credentials' => array('key' => $config->s3_key, 'secret' => $config->s3_secret),
         'region' => $config->s3_region,
-        'version' => AWS_API_VERSION
+        'version' => AWS_API_VERSION,
+            'http'    => [
+                'proxy' => $proxy,
```

Comment:
Also would it be safe to pass an empty proxy? would it be the same as no proxy?

Reply:
I have tested it in the UI, with work as "no proxy". However, it will put a condition there as it will be safer.